kotlinx-coroutines-core/jvm/src/debug/internal/DebugCoroutineInfoImpl.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.debug.internal
+
+import java.lang.ref.*
+import kotlin.coroutines.*
+import kotlin.coroutines.jvm.internal.*
+
+internal const val CREATED = "CREATED"
+internal const val RUNNING = "RUNNING"
+internal const val SUSPENDED = "SUSPENDED"
+
+/**
+ * Internal implementation class where debugger tracks details it knows about each coroutine.
+ */
+internal class DebugCoroutineInfoImpl(
+    context: CoroutineContext?,
+    public val creationStackBottom: CoroutineStackFrame?,
+    @JvmField internal val sequenceNumber: Long
+) {
+    /**
+     * We cannot keep a strong reference to the context, because with the [Job] in the context it will indirectly
+     * keep a reference to the last frame of an abandoned coroutine which the debugger should not be preventing
+     * garbage-collection of. The reference to context will not disappear as long as the coroutine itself is not lost.
```

Comment:
```suggestion
     * garbage-collection of. The reference to context will not disappear as long as the coroutine itself is not lost.
     * Though CoroutineStackFrame potentially can leak the coroutine as well, we do not wrap it into weak reference: 
     * the class is internal, and we pass only artificially-constructed exceptions wrappers as CoroutineStackFrame
```

Reply:
I've actually changed type to make it clear we don't leak anything.